### PR TITLE
Deploy to Mode & Blast

### PR DESCRIPTION
**Pull request by tkporter**
### Description
See https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3669 for the followup with an ISM and IGP update

### CODE REVIEW

File: rust/config/mainnet_config.json [hunk 1]
```diff
@@ -220,6 +220,51 @@
       "storageGasOracle": "0xBF12ef4B9f307463D3FB59c3604F294dDCe287E2",
       "validatorAnnounce": "0x182E8d7c5F1B06201b102123FC7dF0EaeB445a7B"
     },
+    "blast": {
+      "blockExplorers": [
```
Comment (daniel-savu) [inline rust/config/mainnet_config.json:224]: I think this is the api endpoint https://api.blastscan.io/api

File: typescript/infra/config/environments/mainnet3/tokenPrices.json [hunk 2]
```diff
@@ -3,9 +3,11 @@
   "arbitrum": "2914.61",
   "avalanche": "35.71",
   "bsc": "373.19",
+  "blast": "2914.61",
```
Comment (daniel-savu) [inline typescript/infra/config/environments/mainnet3/tokenPrices.json:6]: i'm seeing $14.56 on this site: https://coinranking.com/coin/xUvZdwsex+blast-blast

File: typescript/infra/config/environments/mainnet3/tokenPrices.json [hunk 3]
```diff
@@ -3,9 +3,11 @@
...
   "celo": "0.772619",
   "ethereum": "2914.61",
   "mantapacific": "2914.61",
+  "mode": "2914.61",
```
Comment (daniel-savu) [inline typescript/infra/config/environments/mainnet3/tokenPrices.json:10]: this one looks under $1

File: typescript/infra/scripts/agents/utils.ts [hunk 4]
```diff
@@ -32,6 +32,9 @@ export class AgentCli {
       switch (role) {
         case Role.Validator:
           for (const chain of this.agentConfig.contextChainNames[role]) {
+            if (chain !== 'mode' && chain !== 'blast') {
```
Comment (daniel-savu) [inline typescript/infra/scripts/agents/utils.ts:35]: rm now that we're merging this?

File: typescript/infra/src/deployment/deploy.ts [hunk 5]
```diff
@@ -136,8 +136,14 @@ export async function writeAgentConfig(
       }
 
       const mailbox = core.getContracts(chain).mailbox;
-      const deployedBlock = await mailbox.deployedBlock();
-      return deployedBlock.toNumber();
+      console.log('mailbox', mailbox);
```
Comment (daniel-savu) [inline typescript/infra/src/deployment/deploy.ts:139]: rm?

**Review by daniel-savu — APPROVED** [review]

File: rust/agents/scraper/migration/src/m20230309_000001_create_table_domain.rs [hunk 6]
```diff
@@ -49,8 +49,16 @@ const DOMAINS: &[RawDomain] = &[
     RawDomain {
         name: "base",
         token: "ETH",
-        domain: 8453,
-        chain_id: 8453,
+        domain: 81457,
+        chain_id: 81457,
```
Comment (paulbalaji) [inline rust/agents/scraper/migration/src/m20230309_000001_create_table_domain.rs:53]: why did the base chainid change to blast;s?

File: rust/agents/scraper/migration/src/m20230309_000001_create_table_domain.rs [hunk 7]
```diff
@@ -49,8 +49,16 @@ const DOMAINS: &[RawDomain] = &[
...
+        is_test_net: false,
+        is_deprecated: false,
+    },
+    RawDomain {
+        name: "blast",
+        token: "ETH",
+        domain: 34443,
+        chain_id: 34443,
```
Comment (paulbalaji) [inline rust/agents/scraper/migration/src/m20230309_000001_create_table_domain.rs:61]: why is blast's id that of mode?

File: typescript/infra/config/environments/mainnet3/tokenPrices.json (hunk 2 above)
Comment (paulbalaji) [reply typescript/infra/config/environments/mainnet3/tokenPrices.json:6]: think this is just the gas token prices, iirc blast uses eth for gas (ditto mode below)

File: rust/config/mainnet_config.json (hunk 1 above)
Comment (paulbalaji) [reply rust/config/mainnet_config.json:224]: blastscan: https://docs.blastscan.io/getting-started/endpoint-urls
blastexplorer: https://blastexplorer.io/documentation/etherscan-compatibility/accounts

think we've only got the blastexplorer one defined in metadata rn

File: typescript/infra/config/environments/mainnet3/tokenPrices.json (hunk 3 above)
Comment (tkporter) [reply typescript/infra/config/environments/mainnet3/tokenPrices.json:10]: see https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/3653#discussion_r1579354284

File: rust/config/mainnet_config.json (hunk 1 above)
Comment (tkporter) [reply rust/config/mainnet_config.json:224]: ah good call, thanks

**Pull request merged**